src/plugins/fs/generic.c

```diff
@@ -715,7 +715,8 @@ static gboolean xfs_resize_device (const gchar *device, guint64 new_size, const
                    from the resize is more important so just ignore the
                    unmount error */
                 g_clear_error (&local_error);
-        }
+        } else
+            g_rmdir (mountpoint);
```

Comment:
I think you still need to handle the above cases like ` if (!xfs_info) `  (I'm unable to add comment to this PR since that line wasn't changed)

Reply:
Thanks, I've just moved the `fs_mount` call after the `bd_fs_xfs_get_info` one -- we can get the information we need without mounting, so it will be easier.